hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyCommitResponse.java

```diff
@@ -118,9 +118,12 @@ public RepeatedOmKeyInfo getKeysToDelete() {
 
   protected void updateDeletedTable(OMMetadataManager omMetadataManager,
       BatchOperation batchOperation) throws IOException {
-    if (this.keysToDelete != null) {
+    if (this.keysToDelete != null
+        && this.keysToDelete.getOmKeyInfoList().size() > 0) {
+      String deleteKey = omMetadataManager.getOzoneDeletePathKey(
+          keysToDelete.getOmKeyInfoList().get(0).getObjectID(), openKeyName);
```

Comment:
@szetszwo A tracking JIRA is already raised to handle this separately [HDDS-8467](https://issues.apache.org/jira/browse/HDDS-8467)

Reply:
@sumitagrawl , [HDDS-8467](https://issues.apache.org/jira/browse/HDDS-8467) is for `OBJECT_ID_RECLAIM_BLOCKS`.

But this comment is talking about `getOmKeyInfoList().get(0)`.  They are different problems.

Reply:
BTW, if this pull request has some known problem, we cannot merge it to the master branch even there are JIRAs tracking it.  We cannot introduce bugs to master branch and fix it later.

If there is a need, we may create a feature branch.  Then, we can merge some not-working intermediate code and then fix it before merging the feature branch to master.

Reply:
@szetszwo 
In keyList, there will be possible 2 entry as keysToDelete.getOmKeyInfoList(),
- if VersionEnabled is false and old key is present, it will use old key objectId which will be unique.
- If above is not present for delete, and uncommited block is present, it will use OBJECT_ID_RECLAIM_BLOCKS which have value "0", can can have conflict key., This part will be handled as part of [HDDS-8467](https://issues.apache.org/jira/browse/HDDS-8467)

So this currently do not have issue and one issue will be handled as part of HDDS-8467

Reply:
First of all, I disagree that we can use the first key of the list (i,e, `getOmKeyInfoList().get(0)`) to represent the entire list.  Why not putting a deleteKey for each element in the list?

> ... one issue will be handled as part of [HDDS-8467](https://issues.apache.org/jira/browse/HDDS-8467)

As mentioned previously, we cannot introduce bugs to the master branch and fix it later.

Reply:
Agree with @szetszwo. We might need to handle HDDS-8467, before committing this patch.

Reply:
@szetszwo @aswinshakil Updated... For old version, using obectId of old version, for uncommited block using trxLogIndex.